mmcv/utils/registry.py

```diff
@@ -28,14 +107,57 @@ def __repr__(self):
                      f'items={self._module_dict})'
         return format_str
 
+    @staticmethod
+    def infer_scope():
+        """Infer the scope of registry.
+
+        The name of the package where registry is defined will be returned
+
+        Returns:
+            scope (str): The inferred scope name.
+        """
+        filename = inspect.getmodule(inspect.stack()[2][0]).__name__
+        split_filename = filename.split('.')
+        return split_filename[0]
+
+    @staticmethod
+    def split_scope_key(key):
+        """Split scope and key.
+
+        The first scope will be split from key.
+
+        Examples:
+            >>> Registry.split_scope_key('mmdet.ResNet')
+            'mmdet', 'ResNet'
+            >>> Registry.split_scope_key('ResNet')
+            None, 'ResNet'
+
+        Return:
+            scope (str, None): The first scope.
+            key (str): The remaining key.
+        """
+        split_index = key.find('.')
```

Comment:
What if there are more than one dot in `key`?

Reply:
Handled in a recursive way.